### PR TITLE
Generalize workaround

### LINKED ISSUE #417: Games using SDL-image fail to launch
**Game information**
Shadow of mordor

**Distribution name and version where applicable**
$ cat /etc/os-release
NAME=Fedora
VERSION="30.20190814.0 (Workstation Edition)"
ID=fedora
VERSION_ID=30
VERSION_CODENAME=""
PLATFORM_ID="platform:f30"
PRETTY_NAME="Fedora 30.20190814.0 (Workstation Edition)"
ANSI_COLOR="0;34"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:30"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f30/system-administrators-guide/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=30
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=30
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Workstation Edition"
VARIANT_ID=workstation
OSTREE_VERSION=30.20190814.0

-->

**Flatpak info**

**Problem description**
do not open

log
 https://paste.fedoraproject.org/paste/a24qSmlATA7aZw95GkSCJw/deactivate/5zlXh97pUGG16ietkQgiTsVW3iSxjlVJhbSwCsahwp4Mccfgoy4gtqpFOsNB2A3m

**Does this issue reproduce with native Steam**

### PR DESCRIPTION
**Pull request by nanonyme**
Fixes #417

### CODE REVIEW

**Comment by FastOSlinux** [comment]
build 8403 

$ flatpak run com.valvesoftware.Steam//test
https://github.com/flathub/com.valvesoftware.Steam/wiki/Frequently-asked-questions
Overriding TZ to America/Sao_Paulo
Running Steam on org.freedesktop.platform 19.08.1 64-bit
STEAM_RUNTIME is enabled automatically
......

Loaded Config for Local Override Path for App ID 255220, Controller 0: /var/home/fastos/.local/share/Steam//controller_base/empty.vdf
GameAction [AppID 255220, ActionID 3] : LaunchApp changed task to ProcessingInstallScript with ""
GameAction [AppID 255220, ActionID 3] : LaunchApp changed task to SynchronizingCloud with ""
GameAction [AppID 255220, ActionID 3] : LaunchApp changed task to SiteLicenseSeatCheckout with ""
GameAction [AppID 255220, ActionID 3] : LaunchApp changed task to CreatingProcess with ""
GameAction [AppID 255220, ActionID 3] : LaunchApp waiting for user response to CreatingProcess ""
GameAction [AppID 255220, ActionID 3] : LaunchApp continues with user response "CreatingProcess"
Game update: AppID 255220 "", ProcID 566, IP 0.0.0.0:0
Starting app 255220
Loaded Config for Local Override Path for App ID 255220, Controller 0: /var/home/fastos/.local/share/Steam//controller_base/empty.vdf
>>> Adding process 566 for game ID 255220
GameAction [AppID 255220, ActionID 3] : LaunchApp changed task to WaitingGameWindow with ""
GameAction [AppID 255220, ActionID 3] : LaunchApp changed task to Completed with ""
ERROR: ld.so: object '/var/home/fastos/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/var/home/fastos/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
pid 568 != 567, skipping destruction (fork without exec?)
ERROR: ld.so: object '/var/home/fastos/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/var/home/fastos/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
/var/mnt/ba9bc4fd-8849-4ca0-9e5c-744c157e2184/jogos3/steamapps/common/GRID Autosport/bin/GridAutosport: symbol lookup error: /usr/lib/x86_64-linux-gnu/libSDL2_image-2.0.so.0: undefined symbol: SDL_RWread
>>> Adding process 567 for game ID 255220
>>> Adding process 569 for game ID 255220
Game removed: AppID 255220 "", ProcID 566 
Exiting app 255220

File: resources/freedesktop-sdk.ld.so.blockedlist [hunk 1]
```diff
@@ -1,82 +1,82 @@
...
+*(*/)Sid\ Meier\'s\ Civilization\ VI/GameGuide/libexec/QtWebEngineProcess *(*/)Sid\ Meier\'s\ Civilization\ VI/GameGuide/lib/libXext.so.6
+*(*/)Sid\ Meier\'s\ Civilization\ VI/GameGuide/libexec/QtWebEngineProcess *(*/)Sid\ Meier\'s\ Civilization\ VI/GameGuide/lib/libXfixes.so.3
+*(*/)Sid\ Meier\'s\ Civilization\ VI/GameGuide/libexec/QtWebEngineProcess *(*/)Sid\ Meier\'s\ Civilization\ VI/GameGuide/lib/libXi.so.6
+*(*/)Sid\ Meier\'s\ Civilization\ VI/GameGuide/libexec/QtWebEngineProcess *(*/)Sid\ Meier\'s\ Civilization\ VI/GameGuide/lib/libxml2.so.2
+*(*/)Sid\ Meier\'s\ Civilization\ VI/GameGuide/libexec/QtWebEngineProcess *(*/)Sid\ Meier\'s\ Civilization\ VI/GameGuide/lib/libXrender.so.1
+*(*/)Sid\ Meier\'s\ Civilization\ VI/GameGuide/libexec/QtWebEngineProcess *(*/)Sid\ Meier\'s\ Civilization\ VI/GameGuide/lib/libxslt.so.1
+*(*/)Sid\ Meier\'s\ Civilization\ VI/GameGuide/libexec/QtWebEngineProcess *(*/)Sid\ Meier\'s\ Civilization\ VI/GameGuide/lib/libXtst.so.6
+*(*/)Sid\ Meier\'s\ Civilization\ VI/GameGuide/libexec/QtWebEngineProcess *(*/)Sid\ Meier\'s\ Civilization\ VI/GameGuide/lib/libXxf86vm.so.1
+*(*/)Steam/steamapps/common/*(*/)* *(*/)steam-runtime/+(*/)libSDL2-2.0.so.0.9.0
```
Comment (valentindavid) [inline resources/freedesktop-sdk.ld.so.blockedlist]: If you install a game in a different "content library folder", then "Steam" prefix is not there. You should match "*(/*)/steamaps/common/*(*/)*"

**Review by valentindavid — CHANGES_REQUESTED** [review]

File: resources/freedesktop-sdk.ld.so.blockedlist (hunk 1 above)
Comment (valentindavid) [reply resources/freedesktop-sdk.ld.so.blockedlist]: There are some * that were removed by Mardown.

Comment (nanonyme) [reply resources/freedesktop-sdk.ld.so.blockedlist]: Right, that makes sense. I'll anyway move to use YAML config before continuing with this.

File: resources/gen-proton-blocklist.sh [hunk 2]
```diff
@@ -14,7 +14,7 @@ for proton in "${!PROTON_BLOCK_LIBS[@]}"; do
     for block_lib in ${PROTON_BLOCK_LIBS[$proton]}; do
         for arch_suffix in "64" ""; do
             for wine_bin in "wine${arch_suffix}" "wine${arch_suffix}-preloader"; do
-                echo "${dist_dir}/bin/${wine_bin} ${dist_dir}/lib${arch_suffix}/${block_lib}"
+                echo "*(*/)${dist_dir}/bin/${wine_bin} *(*/)${dist_dir}/lib${arch_suffix}/${block_lib}"
```
Comment (gasinvein) [inline resources/gen-proton-blocklist.sh]: What are these `*(*/)`?

Comment (nanonyme) [reply resources/gen-proton-blocklist.sh]: GNU extension for fnmatch. The first star means "zero or more"

Comment (gasinvein) [reply resources/gen-proton-blocklist.sh]: Are these required now and path component without wildcards won't match?

Comment (nanonyme) [reply resources/gen-proton-blocklist.sh]: Yes, pattern match is now pure fnmatch. It allows wildcards at beginning and end of pattern.

Comment (gasinvein) [reply resources/gen-proton-blocklist.sh]: Then I guess the the config-generator shouldn't escape backslashes.

**Comment by gasinvein** [comment]
@nanonyme Maybe move game-specific stuff out from `blocklists/core.yml`?

**Comment by nanonyme** [comment]
Maybe I should. To be honest, I mostly created that because most games only require one or two libraries blocked so having separate files for everything seemed overkill. But games like Civilization VI will make everything else basically unreadable.

**Comment by nanonyme** [comment]
I'm going to close this as Steam made a working solution that is present in Steam beta.

**Pull request closed without merging**